docs/ocp-deployment.md

```diff
@@ -0,0 +1,863 @@
+export EXAMPLES_DIR="examples/$BASE_NAME"
+export MONITORING_NAMESPACE="openshift-user-workload-monitoring" 
+```
+
+*Note*: the `MONITORING_NAMESPACE` environment variable may need to change depending on your OCP setup.
+
+Then, create the namespace in which the `llm-d` infrastructure will be deployed:
+
+```bash
+kubectl create namespace $NAMESPACE
+```
+
+#### Deploying the Workload-Variant-Autoscaler
+
+Before running the Make target to deploy Workload-Variant-Autoscaler, the `PROMETHEUS_BASE_URL` in the `config/manager/configmap.yaml` must be changed into a valid URL, to be able to connect to Thanos:
+
+```yaml
+# ...
+  PROMETHEUS_BASE_URL: "https://thanos-querier.openshift-monitoring.svc.cluster.local:9091"
+```
+
+**Note**: the Prometheus/Thanos URL may change depending on your OCP setup. You may want to run the following command to find the appropriate Service:
+
+```bash
+kubectl get svc -n openshift-monitoring
+
+kubectl get svc -n openshift-user-workload-monitoring
+```
+
+And then use the appropriate URL associated with the `thanos-querier` Service.
+
+After that, you can deploy the Workload-Variant-Autoscaler using the basic `Make` target:
+
+```sh
+make deploy IMG=ghcr.io/llm-d/workload-variant-autoscaler:latest
+```
+
+Then, you need to deploy the required ConfigMaps for the accelerator costs and the service classes. An example of this configuration can be found in the following command.
+
+```sh
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: ConfigMap
+# This configMap defines the set of accelerators available
+# to the autoscaler to assign to variants
+#
+# For each accelerator, need to specify a (unique) name and some properties:
+# - device is the name of the device (card) corresponding to this accelerator,
+#   it should be the same as the device specified in the node object
+# - cost is the cents/hour cost of this accelerator
+#
+metadata:
+  name: service-classes-config
+  namespace: workload-variant-autoscaler-system
+data:
+  premium.yaml: |
+    name: Premium
+    priority: 1
+    data:
+      - model: default/default
+        slo-tpot: 24
+        slo-ttft: 500
+      - model: llama0-70b
+        slo-tpot: 80
+        slo-ttft: 500
+      - model: unsloth/Meta-Llama-3.1-8B
+        slo-tpot: 9
+        slo-ttft: 1000
+  freemium.yaml: |
+    name: Freemium
+    priority: 10
+    data:
+      - model: granite-13b
+        slo-tpot: 200
+        slo-ttft: 2000
+      - model: llama0-7b
+        slo-tpot: 150
+        slo-ttft: 1500
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: accelerator-unit-costs
+  namespace: workload-variant-autoscaler-system
+data:
+  A100: |
+    {
+    "device": "NVIDIA-A100-PCIE-80GB",
+    "cost": "40.00"
+    }
+  MI300X: |
+    {
+    "device": "AMD-MI300X-192GB",
+    "cost": "65.00"
+    }
+  G2: |
+    {
+    "device": "Intel-Gaudi-2-96GB",
+    "cost": "23.00"
+    }
+  H100: |
+    {
+    "device": "NVIDIA-H100-80GB-HBM3",
+    "cost": "100.0"
+    }
+EOF
+
+```
+
```

Comment:
reminder to add 

```
oc adm policy add-cluster-role-to-user cluster-monitoring-view -z prometheus-adapter -n openshift-user-workload-monitoring
```

and 

```
oc adm policy add-cluster-role-to-user cluster-monitoring-view -z workload-variant-autoscaler-controller-manager -n workload-variant-autoscaler-system
```

somewhere in this guide

Reply:
and here is what the error looks like in the prom-adapter pods if the policy is not applied

```
E0926 17:23:24.977137       1 provider.go:229] unable to update list of all metrics: unable to fetch metrics for query "{__name__=~\"^container_.*\",container!=\"POD\",namespace!=\"\",pod!=\"\"}": bad_response: unknown response code 403
E0926 17:23:24.977878       1 periodic_metric_lister.go:60] unable to update list of all metrics: unable to fetch metrics for query "inferno_desired_replicas{variant_name!=\"\",exported_namespace!=\"\"}": bad_response: unknown response code 4
```